src/components/ItemDisplay.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useEffect } from 'react';
+import logo from '../assets/one.jpg';
+import { format } from 'date-fns';
+
+const ItemDisplay = ({ setShowModal, closeModal, details }) => {
+	const mainRef = useRef(null);
+
+	useEffect(() => {
+		mainRef.current = document.getElementById('modal');
+
+		window.addEventListener('click', function (e) {
+			if (e.target === mainRef.current) {
+				setShowModal(false);
+			}
+		});
+	}, [setShowModal]);
+
+	const convertFromNull =
+		details.dateLastPurchased === null
+			? details.dateCreated
+			: details.dateLastPurchased;
+
+	const convertToDate = (date) => {
+		let dateValue = date.toDate();
+		let day = date.toDate().getDay();
```

Comment:
Why call `date.toDate().getDay()` again when you have dateValue?
```suggestion
		let day = date.getDay();
```

Reply:
This has been updated